plugins/GSdx/GSTextureCache.h

```diff
@@ -29,6 +29,82 @@ class GSTextureCache
 public:
 	enum {RenderTarget, DepthStencil};
 
+	template <class T>
+	struct Element {
+		T  data;
+		uint16 next_index;
+		uint16 prev_index;
+	};
+
+	template <class T>
+	class FastList {
+	public:
+		// Seems like 512 is a reasonable upper bound for the size (ZoE2 goes just below 500), maybe 1024 better
+		static const uint16 SIZE = 512;
+		// The last element of the list has next index equal to END
+		static const uint16 END = SIZE;
+		// m_buffer[0] is the head of the list
+		Element<T> m_buffer[SIZE];
```

Comment:
Nevermind I was joking.
I implemented a automatically growing FastList which does not invalidate iterator simply using realloc() and increasing the size of the two underlying arrays.
Benchmarks shows that there is not much decrease in performance wrt the statically allocated / fixed size one, so the latter is for sure a better choice. I need to clean this PR.